Lib/_ldap.pyi

```diff
@@ -0,0 +1,406 @@
+MOD_INCREMENT: int
+MOD_REPLACE: int
+MSG_ALL: int
+MSG_ONE: int
+MSG_RECEIVED: int
+NO_LIMIT: int
+OPT_API_FEATURE_INFO: int
+OPT_API_INFO: int
+OPT_CLIENT_CONTROLS: int
+OPT_CONNECT_ASYNC: int
+OPT_DEBUG_LEVEL: int
+OPT_DEFBASE: int
+OPT_DEREF: int
+OPT_DESC: int
+OPT_DIAGNOSTIC_MESSAGE: int
+OPT_ERROR_NUMBER: int
+OPT_ERROR_STRING: int
+OPT_HOST_NAME: int
+OPT_MATCHED_DN: int
+OPT_NETWORK_TIMEOUT: int
+OPT_OFF: int
+OPT_ON: int
+OPT_PROTOCOL_VERSION: int
+OPT_REFERRALS: int
+OPT_REFHOPLIMIT: int
+OPT_RESTART: int
+OPT_RESULT_CODE: int
+OPT_SERVER_CONTROLS: int
+OPT_SIZELIMIT: int
+OPT_SUCCESS: int
+OPT_TCP_USER_TIMEOUT: int
+OPT_TIMELIMIT: int
+OPT_TIMEOUT: int
+OPT_URI: int
+OPT_X_KEEPALIVE_IDLE: int
+OPT_X_KEEPALIVE_INTERVAL: int
+OPT_X_KEEPALIVE_PROBES: int
+OPT_X_SASL_AUTHCID: int
+OPT_X_SASL_AUTHZID: int
+OPT_X_SASL_MECH: int
+OPT_X_SASL_NOCANON: int
+OPT_X_SASL_REALM: int
+OPT_X_SASL_SECPROPS: int
+OPT_X_SASL_SSF: int
+OPT_X_SASL_SSF_EXTERNAL: int
+OPT_X_SASL_SSF_MAX: int
+OPT_X_SASL_SSF_MIN: int
+OPT_X_SASL_USERNAME: int
+OPT_X_TLS: int
+OPT_X_TLS_ALLOW: int
+OPT_X_TLS_CACERTDIR: int
+OPT_X_TLS_CACERTFILE: int
+OPT_X_TLS_CERTFILE: int
+OPT_X_TLS_CIPHER: int
+OPT_X_TLS_CIPHER_SUITE: int
+OPT_X_TLS_CRLCHECK: int
+OPT_X_TLS_CRLFILE: int
+OPT_X_TLS_CRL_ALL: int
+OPT_X_TLS_CRL_NONE: int
+OPT_X_TLS_CRL_PEER: int
+OPT_X_TLS_CTX: int
+OPT_X_TLS_DEMAND: int
+OPT_X_TLS_DHFILE: int
+OPT_X_TLS_ECNAME: int
+OPT_X_TLS_HARD: int
+OPT_X_TLS_KEYFILE: int
+OPT_X_TLS_NEVER: int
+OPT_X_TLS_NEWCTX: int
+OPT_X_TLS_PACKAGE: int
+OPT_X_TLS_PEERCERT: int
+OPT_X_TLS_PROTOCOL_MAX: int
+OPT_X_TLS_PROTOCOL_MIN: int
+OPT_X_TLS_PROTOCOL_SSL3: int
+OPT_X_TLS_PROTOCOL_TLS1_0: int
+OPT_X_TLS_PROTOCOL_TLS1_1: int
+OPT_X_TLS_PROTOCOL_TLS1_2: int
+OPT_X_TLS_PROTOCOL_TLS1_3: int
+OPT_X_TLS_RANDOM_FILE: int
+OPT_X_TLS_REQUIRE_CERT: int
+OPT_X_TLS_REQUIRE_SAN: int
+OPT_X_TLS_TRY: int
+OPT_X_TLS_VERSION: int
+PORT: int
+REQ_ABANDON: int
+REQ_ADD: int
+REQ_BIND: int
+REQ_COMPARE: int
+REQ_DELETE: int
+REQ_EXTENDED: int
+REQ_MODIFY: int
+REQ_MODRDN: int
+REQ_SEARCH: int
+REQ_UNBIND: int
+RES_ADD: int
+RES_ANY: int
+RES_BIND: int
+RES_COMPARE: int
+RES_DELETE: int
+RES_EXTENDED: int
+RES_INTERMEDIATE: int
+RES_MODIFY: int
+RES_MODRDN: int
+RES_SEARCH_ENTRY: int
+RES_SEARCH_REFERENCE: int
+RES_SEARCH_RESULT: int
+RES_UNSOLICITED: int
+SASL_AUTOMATIC: int
+SASL_AVAIL: int
+SASL_INTERACTIVE: int
+SASL_QUIET: int
+SCOPE_BASE: int
+SCOPE_ONELEVEL: int
+SCOPE_SUBORDINATE: int
+SCOPE_SUBTREE: int
+SYNC_INFO: str
+TAG_CONTROLS: int
+TAG_EXOP_REQ_OID: int
+TAG_EXOP_REQ_VALUE: int
+TAG_EXOP_RES_OID: int
+TAG_EXOP_RES_VALUE: int
+TAG_LDAPCRED: int
+TAG_LDAPDN: int
+TAG_MESSAGE: int
+TAG_MSGID: int
+TAG_NEWSUPERIOR: int
+TAG_REFERRAL: int
+TAG_SASL_RES_CREDS: int
+TLS_AVAIL: int
+URL_ERR_BADSCOPE: int
+URL_ERR_MEM: int
+VENDOR_VERSION: int
+VERSION: int
+VERSION1: int
+VERSION2: int
+VERSION3: int
+VERSION_MAX: int
+VERSION_MIN: int
+
+class ADMINLIMIT_EXCEEDED(ldap.LDAPError):
```

Comment:
Better move `LDAPError` definition to the top and avoid circular import...

```suggestion
class ADMINLIMIT_EXCEEDED(LDAPError):
```

Reply:
I'd rather leave things like they are in order to keep the `.pyi` file as similar to the original output of `stubgen` as possible...correct me if I'm wrong, but circular imports shouldn't be much of a concern with `.pyi` files?

Reply:
> correct me if I'm wrong, but circular imports shouldn't be much of a concern with .pyi files?

That's true. But I think following `stubgen` output should be a non-goal. stubgen is something a project should only do once, and for the rest of the lifetime, `.pyi` files will be for humans to read and edit. So I would emphasize readability over conformance with stubgen output.

If you want to ensure the `.pyi` doesn't get outdated compared to the implementation, it's much better to run `stubtest` in CI.